drivers/vulkan/vulkan_context.cpp

```diff
@@ -740,13 +740,16 @@ Error VulkanContext::_check_capabilities() {
 
 		device_properties_func(gpu, &physicalDeviceProperties);
 
-		subgroup_capabilities.size = subgroupProperties.subgroupSize;
-		subgroup_capabilities.supportedStages = subgroupProperties.supportedStages;
-		subgroup_capabilities.supportedOperations = subgroupProperties.supportedOperations;
-		// Note: quadOperationsInAllStages will be true if:
-		// - supportedStages has VK_SHADER_STAGE_ALL_GRAPHICS + VK_SHADER_STAGE_COMPUTE_BIT.
-		// - supportedOperations has VK_SUBGROUP_FEATURE_QUAD_BIT.
-		subgroup_capabilities.quadOperationsInAllStages = subgroupProperties.quadOperationsInAllStages;
+		// Be sure not to advertise subgroup support in Vulkan 1.0, where it's not supported.
+		if (vulkan_major >= 1 || vulkan_minor >= 1) {
```

Comment:
Is it valid to call ``vkGetPhysicalDeviceProperties2KHR`` with the ``subgroupProperties`` struct in Vulkan 1.0? I wonder if all of the subgroup stuff here should be hidden in Vulkan 1.0

Reply:
I think we're here in the same situation as we were with another function pointer. I guess if we use `vkGetDeviceProcAddr()` we can rely on `device_properties_func` being null or not, regardless its Vulkan 1.0 or higher. According to the comment at line 710, Vulkan 1.0 may still give a KHR-suffixed version of the function.

Reply:
Not sure if that belongs to the scope of this PR. We can change all the `vkGetInstanceProcAddr()` to `vkGetDeviceProcAddr()` in another one.

Reply:
Ah, right, that makes sense. 

For this code, we can assume it is only run when ``vkGetPhysicalDeviceProperties2KHR`` or ``vkGetPhysicalDeviceProperties2`` are available. Since they may be available through extensions, we could still be on Vulkan 1.0, so we need to avoid advertising subgroup support there, but since the function we are calling will still be ``vkGetPhysicalDeviceProperties2`` we don't have to worry about the struct being there. 

Moving to ``vkGetDeviceProcAddr`` does indeed sound like a good idea, but certainly material for another PR.

Reply:
If we can't rely on `VK_STRUCTURE_TYPE_PHYSICAL_DEVICE_SUBGROUP_PROPERTIES` we should probably also put this inside of an if block:
```
subgroupProperties.sType = VK_STRUCTURE_TYPE_PHYSICAL_DEVICE_SUBGROUP_PROPERTIES;
		subgroupProperties.pNext = nextptr;
		nextptr = &subgroupProperties;
```

else you would assume `subgroupProperties` to be populated correctly even in Vulkan 1.0 ?

Reply:
Yeah the whole thing is Vulkan 1.1 so I would skip including `VK_STRUCTURE_TYPE_PHYSICAL_DEVICE_SUBGROUP_PROPERTIES` alltogether on Vulkan 1.0

See: https://registry.khronos.org/vulkan/specs/1.3-extensions/man/html/VkPhysicalDeviceSubgroupProperties.html

Reply:
I've done some changes. Please check. I've moved the Vulkan version check above and then `subgroupProperties` is checked unconditionally, which is safe since it's zero-initialized.

Reply:
I've had some weird issues especially on Android where it doesn't zero-initialize buffers, but i guess the `{}` in `VkPhysicalDeviceSubgroupProperties subgroupProperties{};` ensure this?

If so agreed, if we skip adding that entry to our chain if it's not supported, then we just end up copying zeroes, which is great because it ensures we clear out the settings actually exposed.